### PR TITLE
doc: Lock Deployment Config

### PR DESCRIPTION
**Pull request by ashokdevtron**

### CODE REVIEW

**Review by abhibhaw — COMMENTED** [review]
This looks great. A suggestion to add this as reference at all the places in docs where we have discussed about deployment template / configs in doc. [Example reference and its child pages](https://docs.devtron.ai/usage/applications/creating-application/deployment-template)

File: docs/user-guide/global-configurations/lock-deployment-config.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+While super-admins can directly edit the locked keys, let's look at a scenario where a user (non-super-admin) tries to edit the same in an [unprotected](../../user-guide/creating-application/config-approval.md) base deployment template.
+
+* User can hide/unhide the locked keys as shown below.
+
+    ![Figure 8: Hiding Locked Keys](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/lock-dt/hide-locked-keys.gif)
+
+{% hint style="info" %}
+<span><img src="https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/lock-dt/mode.jpg" alt="GUI or YAML Mode"></span> <br />
+If you have selected 'Basic' mode instead of 'Advanced (YAML)', only unlocked keys would appear.
```
Comment (uxarya-d) [inline docs/user-guide/global-configurations/lock-deployment-config.md:66]: Even if some keys from the `Basic` view are locked, all keys included in `Basic` view will keep showing. The user can change values from the UI but would not be able to save changes to any locked key.

File: docs/user-guide/global-configurations/lock-deployment-config.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    ![Figure 2: Configure Lock Button](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/lock-dt/lock-deployment-config.jpg)
+
+2. (Optional) Click **Refer Values.YAML** to check which keys you wish to lock.
+
+    ![Figure 3: Values.YAML File](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/lock-dt/values-yaml.jpg)
+
+3. Enter the keys inside the editor on the left-hand side, e.g., `resource.limits`.
+
+    ![Figure 4: Referring Values.YAML File](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/lock-dt/side-by-side.jpg)
```
Comment (uxarya-d) [inline docs/user-guide/global-configurations/lock-deployment-config.md:40]: Suggestion: Can we remove this screenshot, the empty state does not seem to add much value. Also, the two sentences together would be better. (something like below)

Enter the keys inside the editor on the left-hand side, e.g., `autoscaling.MaxReplicas`. Use <a href="https://goessner.net/articles/JsonPath/index.html" target="_blank">JSONpath expressions</a> to enter specific keys, lists, or objects to lock.

**Review by uxarya-d — CHANGES_REQUESTED** [review]
Added comments

**Comment by ashokdevtron** [comment]
> This looks great. A suggestion to add this as reference at all the places in docs where we have discussed about deployment template / configs in doc. [Example reference and its child pages](https://docs.devtron.ai/usage/applications/creating-application/deployment-template)

Done

**Review by abhibhaw — APPROVED** [review]
LGTM

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**